test/integration/endpoints/endpoints_test.go

```diff
@@ -291,6 +294,20 @@ func TestEndpointWithTerminatingPod(t *testing.T) {
 
 	// poll until endpoint for deleted Pod is no longer in Endpoints.
 	if err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		// Ensure that the recently deleted Pod exists but with a deletion timestamp. If the Pod does not exist,
+		// we should fail the test since it is no longer validating against a terminating pod.
```

Comment:
i think a question (that i have here) is whether two subsequent API calls - "delete" then quickly "get" would *always* result in the same behavior; i.e. would it always have the pod present for a while with a deletion timestamp.

presumably yes, but can it flake if "get" gets delayed?

Reply:
I think that Delete() set the deltion timestamp synchronously, if the pod object doesn't exist the test must fail

Reply:
If you want to guarantee that you observe the pod in a state of partial deletion, you *have* to set a finalizer on it.

Reply:
It is easy to add on line 199 above. But then you'll have to remove it manually somewhere in this test.

Reply:
I think this test/check is relying on pod graceful deletion behavior running in an integration test without the node controller or kubelet

Reply:
> I think this test/check is relying on pod graceful deletion behavior running in an integration test without the node controller or kubelet

yes, and there was a longer discussion on it here https://github.com/kubernetes/kubernetes/pull/103621#discussion_r667214593. Do we think this is acceptable or should we be explicit and add a finalizer?

Reply:
Or turn this test into an e2e test so we can actually use `terminationGracePeriod`?

Reply:
Ah, there's no controllers? OK, maybe this won't be racy.